docs/contributor/arch/014-telemetry-self-monitor-storage.md

```diff
@@ -0,0 +1,31 @@
+# 14. Telemetry Self Monitoring Storage
```

Comment:
```suggestion
# 14. Telemetry Self-Monitoring Storage
```
You use a hyphen everywhere else, so I'd add it here too.

Strictly speaking, you need a hyphen when it's an adjective and no hyphen (just blank space) when it's a compound noun ("self monitoring is crucial...") , but I fear that might confuse readers.